pkg/gitops/filter.go

```diff
@@ -0,0 +1,90 @@
+package gitops
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+func (cr *CustomRepo) FilterCommits(author string, since time.Time, until time.Time, resource string, namespace string, name string) ([]object.Commit, error) {
+	var logopts git.LogOptions
+	var filteredCommits []object.Commit
+
+	cr.Mutex.Lock()
+	defer cr.Mutex.Unlock()
+	ref, err := cr.Repo.Head()
+	if err != nil {
+		return filteredCommits, fmt.Errorf("could not get ref head from repository: %w", err)
+	}
+
+	logopts.From = ref.Hash()
+	if !since.IsZero() {
+		logopts.Since = &since
+	}
+	if !until.IsZero() {
+		logopts.Until = &until
+	}
+	var resources []string
+	if resource != "" && namespace == "" && name != "" {
+		return filteredCommits, fmt.Errorf("Error (FilterCommits): cannot provide a resource and a name without a namespace")
+	} else if resource == "" && namespace != "" {
+		resources = []string{"k8s-policies", "antrea-policies", "antrea-cluster-policies", "antrea-tiers"}
```

Comment:
@stanleywbwong @Dhruv-J don't we have this list defined somewhere already as a global var?

Reply:
I looked for it earlier in init.go and client.go, but I couldn't find anything.

Reply:
you can add a function which extracts this info from `gvkDirMap` then

Reply:
Yeah I'll use that instead.